leader-election-impl/src/main/java/com/palantir/paxos/PaxosAcceptorImpl.java

```diff
@@ -63,11 +67,15 @@ public PaxosPromise prepare(long seq, PaxosProposalId pid) {
 
             // nack
             if (oldState != null && pid.compareTo(oldState.lastPromisedId) < 0) {
+                leaderLog.debug("Refused proposal request for seq #" + seq + " with ID " + pid.getNumber() +
+                        " as we've already made a promise with a greater pid (" + oldState.lastPromisedId + ")");
                 return new PaxosPromise(oldState.lastPromisedId);
             }
 
             // allow for the same propose to be repeated and return the same result.
             if (oldState != null && pid.compareTo(oldState.lastPromisedId) == 0) {
+                leaderLog.debug("Refused proposal request for seq #" + seq + " with ID " + pid.getNumber() +
```

Comment:
Makes sense. I'll add the leader UUID from the lastAcceptedValue to the log though so that it doesn't look like we're agreeing to accept a proposal for a _different_ value on that sequence number.